### PR TITLE
Fix for #3294 - throw UiRequiredException when WAM cannot get a trans…

### LINKED ISSUE #3294: [Bug] AcquireTokenSilent for OperatingSystemAccount can prompt.
MSAL 4.41 with WAM enabled 

We started to see WAM prompts when acquiring the operating system account silently. 
In all the cases so far the user was signed into windows with an MSA. 

We call 

await pca.AcquireTokenSilent(new string[] { defaultConfiguration.GraphEndpoint.AbsoluteUri + ".default" },PublicClientApplication.OperatingSystemAccount)

This ends up calling 
WebAuthenticationCoreManagerInterop.RequestTokenForWindowAsync

One interesting thing is the documentation for this API indicates it MAY prompt.  
WebAuthenticationCoreManager.RequestTokenAsync Method (Windows.Security.Authentication.Web.Core) - Windows UWP applications | Microsoft Docs
Asynchronously requests a token from a web account provider. If necessary, the user is prompted to enter their credentials.

Microsoft.Identity.Client.Desktop.dll!Microsoft.Identity.Client.Platforms.WebAuthenticationCoreManagerInterop.RequestTokenForWindowAsync(System.IntPtr hWnd, Windows.Security.Authentication.Web.Core.WebTokenRequest request) Line 19
Microsoft.Identity.Client.Desktop.dll!Microsoft.Identity.Client.Platforms.Features.WamBroker.WamProxy.RequestTokenForWindowAsync(System.IntPtr _parentHandle, Windows.Security.Authentication.Web.Core.WebTokenRequest webTokenRequest) Line 73 C#
Microsoft.Identity.Client.Desktop.dll!Microsoft.Identity.Client.Platforms.Features.WamBroker.WamBroker.AcquireTokenSilentDefaultUserPassthroughAsync(Microsoft.Identity.Client.Internal.Requests.AuthenticationRequestParameters authenticationRequestParameters, Windows.Security.Credentials.WebAccountProvider defaultAccountProvider) Line 730          C#
Microsoft.Identity.Client.Desktop.dll!Microsoft.Identity.Client.Platforms.Features.WamBroker.WamBroker.AcquireTokenSilentDefaultUserAsync(Microsoft.Identity.Client.Internal.Requests.AuthenticationRequestParameters authenticationRequestParameters, Microsoft.Identity.Client.ApiConfig.Parameters.AcquireTokenSilentParameters acquireTokenSilentParameters) Line 673              C#
[Completed] Microsoft.Identity.Client.Desktop.dll!Microsoft.Identity.Client.Platforms.Features.WamBroker.WebAccountProviderFactory.GetDefaultProviderAsync() Line 29    C#
[Async] Microsoft.Identity.Client.dll!Microsoft.Identity.Client.Internal.Requests.BrokerSilentStrategy.SendTokenRequestToBrokerAsync() Line 78                C#
[Async] Microsoft.Identity.Client.dll!Microsoft.Identity.Client.Internal.Requests.BrokerSilentStrategy.ExecuteAsync(System.Threading.CancellationToken cancellationToken) Line 49     C#
[Async] Microsoft.Identity.Client.dll!Microsoft.Identity.Client.Internal.Requests.Silent.SilentRequest.ExecuteAsync(System.Threading.CancellationToken cancellationToken) Line 75     C#
[Async] Microsoft.Identity.Client.dll!Microsoft.Identity.Client.Internal.Requests.RequestBase.RunAsync(System.Threading.CancellationToken cancellationToken) Line 88     C#
[Async] Microsoft.Identity.Client.dll!Microsoft.Identity.Client.ApiConfig.Executors.ClientApplicationBaseExecutor.ExecuteAsync(Microsoft.Identity.Client.ApiConfig.Parameters.AcquireTokenCommonParameters commonParameters, Microsoft.Identity.Client.ApiConfig.Parameters.AcquireTokenSilentParameters silentParameters, System.Threading.CancellationToken cancellationToken) Line 41        C#
[Async] Microsoft.Developer.IdentityService.exe!Microsoft.Developer.IdentityService.StorageService.OsTelemetryLogger.PostAccountsSingleSignOnTelemetryAsync.__GetSignedWindowsAccountId|0_2(Microsoft.Identity.Client.IPublicClientApplication pca, System.Diagnostics.TraceSource tracelogger) Line 56   C#

Expectation is that a silent call should be silent and not prompt.

### PR DESCRIPTION
**Pull request by bgavrilMS**
…fer token in MSA-PT

Fixes #3294

**Changes proposed in this request**

**Testing**

**Performance impact**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs [hunk 1]
```diff
@@ -711,6 +711,13 @@ private async Task<MsalTokenResponse> AcquireTokenSilentDefaultUserPassthroughAs
         {
             var transferToken = await _msaPassthroughHandler.TryFetchTransferTokenSilentDefaultAccountAsync(authenticationRequestParameters, defaultAccountProvider).ConfigureAwait(false);
 
+            if (string.IsNullOrEmpty(transferToken))
+            {
+                throw new MsalUiRequiredException(
+                    MsalError.InteractionRequired,
+                    "Found an MSA account, but could not retrieve a transfer token for it.");
```
Comment (jmprieur) [inline src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs:718]: Maybe you want to mention WAM in the string?

```suggestion
                    "Found an MSA account, but could not retrieve a transfer token for it when calling WAM.");
```

Would it be useful to have an aka.ms link to describe what the developer should do?

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs:718]: There is nothing the developer can do, except to pop-up UI via AcquireTokenInteractive. I'll update the error message.

File: tests/Microsoft.Identity.Test.Unit/BrokerTests/WamTests.cs [hunk 2]
```diff
@@ -505,6 +505,65 @@ public async Task ATS_DefaultAccount_Passthrough_Async()
             }
         }
 
+
+        [TestMethod]
+        public async Task ATS_DefaultAccount_NoTransferToken_Passthrough_Async()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/BrokerTests/WamTests.cs:510]: Add ThrowsException to the test name?

**Review by pmaytak — COMMENTED** [review]
What is the current behavior? What is the new behavior? What are the benefits of the new behavior?
- ATS for default OS MSA account in MSA-PT scenario -> can show a UI.
- Same scenario but now instead of UI showing up whenever, dev can catch the UiRequired exception and not show the UI?

**Review by SameerK-MSFT — APPROVED** [review]
LGTM. Liked Peter's suggestion for the test name.

**Review by neha-bhargava — APPROVED** [review]

**Pull request merged**